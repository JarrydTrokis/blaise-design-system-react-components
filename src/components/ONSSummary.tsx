import { Data } from "react-csv/components/CommonPropTypes";
import { FormatTitle } from "../utilities/TextFormatting";
import React, { ReactElement } from "react";

type Group = {
    title: string
    records: Record<string, any>
}
class GroupedSummary {
    groups: Group[];

    constructor(groups: Group[]) {
        this.groups = groups;
    }

    csv(): Data {
        const records: Data = [];
        const row: Record<string, any> = {};
        for (const group of this.groups) {
            for (const record in group.records) {
                row[record] = group.records[record];
            }
        }
        records.push(row);
        return records;
    }
}

interface SummaryItemProps {
    fieldName: string
    fieldValue: any
}

function SummaryItemRow({ fieldName, fieldValue }: SummaryItemProps): ReactElement {
    return (
        <tbody className="summary__item">
            <tr className="summary__row summary__row--has-values">
                <td className="summary__item-title">
                    <div className="summary__item--text">
                        {FormatTitle(fieldName)}
                    </div>
                </td>
                <td className="summary__values" colSpan={2}>
                    {fieldValue}
                </td>
            </tr>
        </tbody>
    );
}

interface SummaryGroupTableProps {
    groupedSummary: GroupedSummary
}

function SummaryGroupTable({ groupedSummary }: SummaryGroupTableProps): ReactElement {
    const elementList: ReactElement[] = [];
    for (const group of groupedSummary.groups) {
        elementList.push(
            <h3 className="summary__group-title">{group.title}</h3>
        );
        const tableFieldsList: ReactElement[] = [];
        for (const field in group.records) {
            tableFieldsList.push(
                <SummaryItemRow fieldName={field} fieldValue={group.records[field]} />
            );
        }
        elementList.push(
            <table className="summary__items">
                {tableFieldsList}
            </table>
        );
    }
    return (
        <>
            {elementList}
        </>
    );
}

export type { Group, SummaryItemProps, SummaryGroupTableProps };
export { GroupedSummary, SummaryItemRow, SummaryGroupTable };
