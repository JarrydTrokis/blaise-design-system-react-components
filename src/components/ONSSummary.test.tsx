import React from "react";
import { SummaryGroupTable, SummaryItemRow, GroupedSummary } from "./ONSSummary";
import { render } from "@testing-library/react";

describe("ONS Summary Group Table test", () => {
    const groupedSummary = new GroupedSummary([{ title: "test", records: { foo: "bar" } }]);

    it("matches Snapshot", () => {
        expect(render(<SummaryGroupTable groupedSummary={groupedSummary} />)).toMatchSnapshot();
    });

    it("should render correctly", () => {
        expect(render(<SummaryGroupTable groupedSummary={groupedSummary} />)).toBeDefined();
    });
});

describe("ONS Summary Item Row test", () => {

    it("matches Snapshot", () => {
        expect(render(<SummaryItemRow fieldName="foo" fieldValue="bar" />)).toMatchSnapshot();
    });

    it("should render correctly", () => {
        expect(render(<SummaryItemRow fieldName="foo" fieldValue="bar" />)).toBeDefined();
    });
});

describe("GroupedSummary as CSV tests", () => {
    const groupedSummary = new GroupedSummary([
        { title: "test", records: { foo: "bar" } },
        { title: "test2", records: { fwibble: "fish" } },
    ]);

    it("converts a grouped summary to a csv type object", () => {
        expect(groupedSummary.csv()).toEqual([{
            foo: "bar",
            fwibble: "fish",
        }]);
    });
});
