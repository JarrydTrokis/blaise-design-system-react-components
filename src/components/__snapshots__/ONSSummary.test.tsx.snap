// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Summary Group Table test matches Snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "summary__group-title",
    },
    "children": Array [
      Node {
        "data": "test",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "h3",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Node {
      "attribs": Object {
        "class": "summary__items",
      },
      "children": Array [
        Node {
          "attribs": Object {
            "class": "summary__item",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "summary__row summary__row--has-values",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "summary__item-title",
                  },
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "class": "summary__item--text",
                      },
                      "children": Array [
                        Node {
                          "data": "Foo",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "td",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "class": "summary__values",
                      "colspan": "2",
                    },
                    "children": Array [
                      Node {
                        "data": "bar",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "td",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "colspan": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "colspan": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Node {
                  "attribs": Object {
                    "class": "summary__values",
                    "colspan": "2",
                  },
                  "children": Array [
                    Node {
                      "data": "bar",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "td",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "summary__item-title",
                    },
                    "children": Array [
                      Node {
                        "attribs": Object {
                          "class": "summary__item--text",
                        },
                        "children": Array [
                          Node {
                            "data": "Foo",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "td",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "colspan": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "colspan": undefined,
                  },
                },
              ],
              "name": "tr",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "tbody",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
      ],
      "name": "table",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": Node {
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": [Circular],
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "parent": Node {
      "children": Array [
        [Circular],
        Node {
          "attribs": Object {
            "class": "summary__items",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "summary__item",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "summary__row summary__row--has-values",
                  },
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "class": "summary__item-title",
                      },
                      "children": Array [
                        Node {
                          "attribs": Object {
                            "class": "summary__item--text",
                          },
                          "children": Array [
                            Node {
                              "data": "Foo",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "td",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Node {
                        "attribs": Object {
                          "class": "summary__values",
                          "colspan": "2",
                        },
                        "children": Array [
                          Node {
                            "data": "bar",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "td",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "colspan": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "colspan": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    Node {
                      "attribs": Object {
                        "class": "summary__values",
                        "colspan": "2",
                      },
                      "children": Array [
                        Node {
                          "data": "bar",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "td",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": Node {
                        "attribs": Object {
                          "class": "summary__item-title",
                        },
                        "children": Array [
                          Node {
                            "attribs": Object {
                              "class": "summary__item--text",
                            },
                            "children": Array [
                              Node {
                                "data": "Foo",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "div",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                        ],
                        "name": "td",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "colspan": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "colspan": undefined,
                      },
                    },
                  ],
                  "name": "tr",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "tbody",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "table",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "1": Node {
    "attribs": Object {
      "class": "summary__items",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "summary__item",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "summary__row summary__row--has-values",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "summary__item-title",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "class": "summary__item--text",
                    },
                    "children": Array [
                      Node {
                        "data": "Foo",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "td",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "class": "summary__values",
                    "colspan": "2",
                  },
                  "children": Array [
                    Node {
                      "data": "bar",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "td",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "colspan": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "colspan": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Node {
                "attribs": Object {
                  "class": "summary__values",
                  "colspan": "2",
                },
                "children": Array [
                  Node {
                    "data": "bar",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "td",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "summary__item-title",
                  },
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "class": "summary__item--text",
                      },
                      "children": Array [
                        Node {
                          "data": "Foo",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "td",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "colspan": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "colspan": undefined,
                },
              },
            ],
            "name": "tr",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "tbody",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "table",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        Node {
          "attribs": Object {
            "class": "summary__group-title",
          },
          "children": Array [
            Node {
              "data": "test",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h3",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": Node {
      "attribs": Object {
        "class": "summary__group-title",
      },
      "children": Array [
        Node {
          "data": "test",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "h3",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": Node {
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": null,
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`ONS Summary Item Row test matches Snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "summary__item",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "summary__row summary__row--has-values",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "summary__item-title",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "summary__item--text",
                },
                "children": Array [
                  Node {
                    "data": "Foo",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "td",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "summary__values",
                "colspan": "2",
              },
              "children": Array [
                Node {
                  "data": "bar",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "td",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "colspan": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "colspan": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Node {
            "attribs": Object {
              "class": "summary__values",
              "colspan": "2",
            },
            "children": Array [
              Node {
                "data": "bar",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "td",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "summary__item-title",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "summary__item--text",
                  },
                  "children": Array [
                    Node {
                      "data": "Foo",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "td",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "colspan": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "colspan": undefined,
            },
          },
        ],
        "name": "tr",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "tbody",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
